instructor/common.py

```diff
@@ -0,0 +1,604 @@
+from django.shortcuts import render_to_response, redirect
+from django.template import RequestContext
+from django import forms
+from django.forms.models import modelformset_factory, inlineformset_factory
+from django.forms.models import modelform_factory
+from instructor import models
+from instructor import compiler
+from django.contrib.auth.models import User
+import datetime
+
+from backend.models import Assignment, StudentAssignment
+
+
+def courses(request):
+    user = request.user
+    message = ''
+    CourseFormSet = modelformset_factory(models.Course, extra=1, fields=['name', 'code'], can_delete=True)
+    if request.method == "POST":
+        formset = CourseFormSet(request.POST)
+        if ( formset.is_valid()):
+            message = "Thank you"
+            entries = formset.save(commit=False)
+            for form in entries:
+                form.owner = request.user
+                form.save()
+        else:
+            message = "Something went wrong"
+
+    return render_to_response('instructor/courses_new.html',
+                              {'formset': CourseFormSet(queryset=models.Course.objects.filter(owner=user)),
+                               'message': message},
+                              context_instance=RequestContext(request))
+
+
+def course_delete(request, pk):
+    print "DELETE {}".format(pk)
+    models.Course.objects.get(id=pk).delete()
+    return redirect('common_course')
+
+
+def get_account_type(request):
+    account_type = ''
+    if request.user.id and len(request.user.groups.all()) > 0:
+        account_type = request.user.groups.all()[0].name
+    return account_type
+
+
+def assignments(request):
```

Comment:
A minimal docstring would be:
```python
"""
View to show all course builder assignments.
"""
```
A more involved example would be:
```
View to show all course builder assignments.

Args:
    request (django.http.HttpRequest): User request.
Returns:
    django.http.HttpResponse: A rendered template of assignments.
```

https://pypi.python.org/pypi/sphinxcontrib-napoleon for some more examples.